### ISSUE TITLE
Manage focus when selecting image link option from block toolbar

### ISSUE THREAD

**Issue by jeryj** [WordPress/gutenberg#58567]
### Description

When setting a link option within the image block toolbar, focus is not intentionally placed, so it is not visible. To be consistent, focus should be placed to the first focusable element within the popover.

### Step-by-step reproduction instructions

1. Have an image in the editor
2. Select the image
3. Go to the block toolbar
4. Arrow to the link button
5. Enter
6. Select Expand on Click or similar
7. Focus is not visible
8. Press Tab
9. Focus is now on the URL bar (in Chrome)

### Screenshots, screen recording, code snippet

https://github.com/WordPress/gutenberg/assets/967608/237fdb48-1ef4-4e8a-b45e-03fb6a195a3d

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by t-hamano** [comment]
Maybe this is a similar issue to #58509?

**Comment by jeryj** [comment]
Yeah, I bet that's the same. Thanks for catching!
